src/axom/inlet/SphinxDocWriter.cpp

```diff
@@ -0,0 +1,98 @@
+#include "axom/inlet/SphinxDocWriter.hpp"
+#include <iostream>
+#include <assert.h>
+#include "axom/slic.hpp"
+
+namespace axom
+{
+namespace inlet
+{
+ 
+SphinxDocWriter::SphinxDocWriter(const std::string& fileName, axom::sidre::Group* root) {
+  SLIC_ASSERT_MSG(root != nullptr, "Sidre Group is null pointer");
+  m_sidreRootGroup = root;
+  m_fileName = fileName;
+}
+
+void SphinxDocWriter::writeDocuments(axom::sidre::Group* sidreGroup) {
+  m_outFile.open(m_fileName);
+  if (sidreGroup->getName() == "") {
+    writeTitle("Untitled");
+  }  else {
+    writeTitle(sidreGroup->getName());

```

Comment:
Suggestion:  Rather then keeping a file open and writing to it directly, it might be cleaner/more efficient to use an ``ostringstream``  for the output and then only open and dump a file at the end.

This might be more resilient to crashes since you won't have to worry (as much) about partially written files.